### DISCUSSION TITLE
Why i use useSelector cannot get data?

### DISCUSSION THREAD

**Discussion by tytom2003** [reduxjs/redux-toolkit#4670] [General]
i used useSelector, but it can't get data. 
My program part:

const ordermgmtAdapter = createEntityAdapter()

const initialState = ordermgmtAdapter.getInitialState()

export const {
    useGetOrdersQuery,
    useUpdateOrderMutation,
} = OrderMgmtApiSlice

export const selectOrderResult = OrderMgmtApiSlice.endpoints.getOrders.select();

const selectOrdersData = createSelector(
    selectOrderResult,
    orderResult => orderResult.data
)

export const {
    selectAll: selectAllOrders,
} = ordermgmtAdapter.getSelectors(state => selectOrdersData(state) ?? initialState)

in other compoent, i used const Orders = useSelector(selectAllOrders); to get data. It cannot get data,
but i used below. orderdata can get data. 

const {
        data: orderdata,
        isSuccess,
        isLoading,
        isError,
        error,
        refetch
      } = useGetOrderQuery()

my progam using package:
"@reduxjs/toolkit": "^2.2.8",
"react-redux": "^9.1.2",
"redux-persist": "^6.0.0",
nodejs: 20.11.1

**Comment by EskiMojo14** [comment]
useSelector only selects state already in store, it won't kick off any fetching processes. you should continue to use the useQuery hook for that.

**Reply by tytom2003** [reply]
if i want to use useUpdateOrderMutation to update data, it can't view new update data at once? 
Thank you very much.

**Reply by EskiMojo14** [reply]
Your choices for making sure the query updates after a mutation are [automated refetching](https://redux-toolkit.js.org/rtk-query/usage/automated-refetching) using tags, or [manual cache updates](https://redux-toolkit.js.org/rtk-query/usage/manual-cache-updates).

**Reply by tytom2003** [reply]
i solve auto update data.
